test/metadata.jl

```diff
@@ -0,0 +1,3065 @@
+            @test collect(colmetadatakeys(res, :c)) == ["name"]
+            @test colmetadata(res, :c, "name") == "c"
+            @test isempty(colmetadatakeys(res, :b))
+            @test collect(colmetadatakeys(res, :a)) == ["name"]
+            @test colmetadata(res, :a, "name") == "a"
+            @test isempty(colmetadatakeys(res, :x1))
+        end
+        for df in (copy(refdf), @view copy(refdf)[:, :])
+            res = fun(groupby(df, :id), :c => copy => AsTable, ungroup=ug)
+            res = parent(parent(res))
+            @test collect(metadatakeys(res)) == ["name"]
+            @test metadata(res, "name") == "refdf"
+            @test collect(colmetadatakeys(res, :id)) == ["name"]
+            @test colmetadata(res, :id, "name") == "id"
+            @test collect(colmetadatakeys(res, :c)) == ["name"]
+            @test colmetadata(res, :c, "name") == "c"
+            @test isempty(colmetadatakeys(res, :b))
+            @test collect(colmetadatakeys(res, :a)) == ["name"]
+            @test colmetadata(res, :a, "name") == "a"
+            @test isempty(colmetadatakeys(res, :x1))
+        end
+        for df in (copy(refdf), @view copy(refdf)[:, :])
+            res = fun(groupby(df, :id), :c => identity => [:a], ungroup=ug)
+            res = parent(parent(res))
+            @test collect(metadatakeys(res)) == ["name"]
+            @test metadata(res, "name") == "refdf"
+            @test collect(colmetadatakeys(res, :id)) == ["name"]
+            @test colmetadata(res, :id, "name") == "id"
+            @test collect(colmetadatakeys(res, :c)) == ["name"]
+            @test colmetadata(res, :c, "name") == "c"
+            @test isempty(colmetadatakeys(res, :b))
+            @test isempty(colmetadatakeys(res, :a))
+        end
+        for df in (copy(refdf), @view copy(refdf)[:, :])
+            res = fun(groupby(df, :id), :c => copy => [:a], ungroup=ug)
+            res = parent(parent(res))
+            @test collect(metadatakeys(res)) == ["name"]
+            @test metadata(res, "name") == "refdf"
+            @test collect(colmetadatakeys(res, :id)) == ["name"]
+            @test colmetadata(res, :id, "name") == "id"
+            @test collect(colmetadatakeys(res, :c)) == ["name"]
+            @test colmetadata(res, :c, "name") == "c"
+            @test isempty(colmetadatakeys(res, :b))
+            @test isempty(colmetadatakeys(res, :a))
+        end
+        for df in (copy(refdf), @view copy(refdf)[:, :])
+            res = fun(groupby(df, :id), identity, ungroup=ug)
+            res = parent(parent(res))
+            @test collect(metadatakeys(res)) == ["name"]
+            @test metadata(res, "name") == "refdf"
+            @test collect(colmetadatakeys(res, :id)) == ["name"]
+            @test colmetadata(res, :id, "name") == "id"
+            @test isempty(colmetadatakeys(res, :a))
+            @test isempty(colmetadatakeys(res, :b))
+            @test isempty(colmetadatakeys(res, :c))
+        end
+    end
+end
```

Comment:
Is `combine(f, gd)` tested somewhere?

Reply:
it was tested with `fun(groupby(df, :id), identity, ungroup=ug)`, but I added additional test to make it more visually clear.

Reply:
`identity` is quite a special case. Are there tests for a case where the returned data frame for each group has no metadata or different metadata from the input?

Reply:
I will add such tests. However, in general, it is not a special case from the point of view of logic of the code (i.e. we just drop all metadata from whatever is returned). But I agree that this is something that should be tested.

Reply:
Added more tests.